lib/unhangout-routes.js

```diff
@@ -788,6 +795,136 @@ module.exports = {
         //
 
+        app.get("/followup/event/:id/participant_:participantCount", function(req, res) {
+
+            var event = db.events.get(req.params.id);
+
+            var eventTitle = event.get("title");
+            var history = event.get("history");
+            var participantCount = req.params.participantCount; 
+
+            var allUserIds = [];
+            allUserIds.push(null);
+
+            _.each(history.event, function(elapsedTime, userId) {
+                
+                //Get the user object for a specific ID 
+                var user = db.users.get(userId);
+
+                // Find all userIDs for users who shared sessions with me at this event.
+                var userIds = event.getUserIdsSharingSessionsWith(userId);
+                
+                // get user objects for the IDs.
+                var users = _.map(userIds, function(userId) { 
+                    return db.users.get(userId); 
+                });
+
+                allUserIds.push({ user: user, users: users });
+
+            });
+
+            var allUserIdsCount = Object.keys(allUserIds).length - 1;
+
+            res.render("email_preview_template.ejs", {
+                eventTitle: eventTitle,
+                participantCount: participantCount,
+                allUserIds: allUserIds,
+                allUserIdsCount: allUserIdsCount,
+                event: event
+            });
+        });
+
+        app.post("/followup/event/:id/sent/", function(req, res) {
+
+            var event = db.events.get(req.params.id);
+            var eventTitle = event.get("title");
+            var history = event.get("history");
+
+            var allUserIds = [];
+
+            _.each(history.event, function(elapsedTime, userId) {
+                
+                //Get the user object for a specific ID 
+                var user = db.users.get(userId);
+
+                // Find all userIDs for users who shared sessions with me at this event.
+                var userIds = event.getUserIdsSharingSessionsWith(userId);
+                
+                // get user objects for the IDs.
+                var users = _.map(userIds, function(userId) { 
+                    return db.users.get(userId); 
+                });
+
+                allUserIds.push({ user: user, users: users });
+
+            }); 
+
+            var allUserIdsCount = Object.keys(allUserIds).length;
+            var mandrill_client = new mandrill.Mandrill('PLDkuaJHZNqb2iuhLSGzYQ');
+
+            var message = {}; 
+            var emailsSent = 0;
+
+            //Execute a loop to send emails
+            for (var participantCount = 0; participantCount < allUserIdsCount; participantCount++) {
+                    
+                    res.render("email_template.ejs", {
+                    eventTitle: eventTitle,
+                    allUserIds: allUserIds,
+                    participantCount: participantCount
+                
+                }, function(err, html) {
```

Comment:
This call will be problematic within a for-loop with async call because the enclosed ``participantCount`` variable will change.  See http://www.mennovanslooten.nl/blog/post/62 for an explanation.  The result you will see is that the last participant will receive every email.

Easiest solution: use ``_.each`` or ``Array.forEach`` instead of a for loop.  Alternatively, wrap the body of the for loop in an anonymous function.